### PR TITLE
fix issue 20562 returning __rvalue should move-construct the NRVO value

### PR DESCRIPTION
**Pull request by WalterBright**
Fixing https://github.com/dlang/dmd/issues/20562

### CODE REVIEW

File: compiler/test/runnable/rvalue1.d [hunk 1]
```diff
@@ -139,6 +139,57 @@ void test7()
...
+{
+    int a,b;
+    this(S8) { printf("this(S)\n"); b = 4; }
+    this(ref S8) { printf("this(ref S)\n"); assert(0); }
+}
+
+
+S8 returnRval(ref S8 arg)
+{
+static if (0)
```
Comment (thewilsonator) [inline compiler/test/runnable/rvalue1.d:155]: what is with these static if branches?

Comment (WalterBright) [reply compiler/test/runnable/rvalue1.d:155]: Variations on the theme. I used them to try to figure out the best way to resolve the bug. In the comment you'll see the code generated for them.

**Comment by WalterBright** [comment]
@TurkeyMan see, I didn't forget!

**Comment by TurkeyMan** [comment]
🎉

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**